### PR TITLE
Improvments on features

### PR DESCRIPTION
**Pull request by animeshgarai09**
Added a few improvments. new video data added and minor bug fixes.

### CODE REVIEW

File: src/store/providers/auth-provider.js [hunk 1]
```diff
@@ -95,7 +101,14 @@ const AuthProvider = ({ children }) => {
                 encodedToken: token
             }
         })
-        setWatchLaterState(user.watchlater)
+        setWatchLaterState({ watchlater: user.watchlater, count: user.watchlater.length })
+        setLikesState({ likes: user.likes, count: user.likes.length })
+        setHistoryState({ history: user.history, count: user.history.length })
+        // console.log(user.playlists)
+        // PlaylistDispatch({
```
Comment (MayankKumar10) [inline src/store/providers/auth-provider.js]: remove comments

File: src/store/providers/like-provider.js [hunk 2]
```diff
@@ -49,7 +48,7 @@ const LikesProvider = ({ children }) => {
                 authorization: encodedToken,
             },
         }).then((response) => {
-            setLikesState({ ...LikesState, likes: response.data.likes });
+            setLikesState({ likes: response.data.likes, count: response.data.likes.length });
         }).catch(err => {
             console.log(err);
```
Comment (MayankKumar10) [inline src/store/providers/like-provider.js:53]: remove console log

File: src/store/providers/playlist-provider.js [hunk 3]
```diff
@@ -67,15 +67,17 @@ const PlaylistProvider = ({ children }) => {
...
             })
-            if (PlaylistState.selectedPlaylistId === null) {
-                PlaylistDispatch({
-                    type: "SET_SELECTED_PLAYLIST",
-                    payload: response.data.playlists[0]._id
-                })
+            if (PlaylistState.playlists.length === 0) {
+                navigate(`/playlists/${response.data.playlists[0]._id}`)
+                // PlaylistDispatch({
+                //     type: "SET_SELECTED_PLAYLIST",
```
Comment (MayankKumar10) [inline src/store/providers/playlist-provider.js]: remove comments

**Review by MayankKumar10 — APPROVED** [review]
Nice UI, Look into these Suggestions everything else looks good to me you can merge it.

File: src/pages/by category/by-category.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+import { useParams } from "react-router-dom"
+import { useVideos } from "@providers"
+import { VideoCard } from "@components"
+
+const ExploreByCategory = () => {
+    const { category } = useParams()
+    const { VideoState: { trending, documentary, gardening } } = useVideos()
+
+    const getVideos = () => {
+        if (category === "trending") {
```
Comment (foolhardy21) [inline src/pages/by category/by-category.js:12]: suggestion - you can use `switch` for multiple conditions.

File: src/pages/by category/by-category.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+            return mapVideoFromArray(trending)
+        } else if (category === "documentaries") {
+            return mapVideoFromArray(documentary)
+        } else if (category === "basic-gardening") {
+            return mapVideoFromArray(gardening)
+        }
+    }
+
+    const mapVideoFromArray = (arr) => {
```
Comment (foolhardy21) [inline src/pages/by category/by-category.js:21]: suggestion - you can trim this function further by removing `return` and `{}`

**Review by foolhardy21 — APPROVED** [review]
rest of the code looks good.

**Pull request merged**